src/aggregator/client/conn.go

```diff
@@ -152,6 +168,55 @@ func (c *connection) Close() {
 	c.mtx.Unlock()
 }
 
+func (c *connection) isTLSConfigCacheValid() bool {
+	if c.tlsConfigCache == nil || c.certsTTLTicker == nil {
```

Comment:
This logic seems duplicated across client and server--consider factoring out a class that handles the cert updates for you under the hood.

Reply:
Also just as a sanity check--is there a library out there that does this already? Of course, given that it's a security thing, we'd probably want to make doubly sure it's reputable.